BasicTypesProxy.cxx

```diff
@@ -24,6 +24,91 @@ namespace
       return x == y;
     }
   }
+
+  /// Helper class to track inf/nans encountered.
+  class InfNanTable
+  {
+    private:
+      struct Encounters
+      {
+        std::size_t count = 0;  /// total number of times this var saw a NaN/inf
+        std::string firstFile;  /// first file a NaN/inf was seen in
+        std::size_t firstEntry = std::numeric_limits<std::size_t>::max(); /// entry number within firstFile where the first NaN/inf was seen
+      };
+
+    public:
+      ~InfNanTable() { EmitTable(); }
+
+      void LogInf(const std::string& varPath, const std::string& file, std::size_t entry)  { Log(fInfEncounters, varPath, file, entry); CheckAbort(); };
+      void LogNaN(const std::string& varPath, const std::string& file, std::size_t entry)  { Log(fNaNEncounters, varPath, file, entry); CheckAbort(); };
+
+      void EmitTable(std::ostream & stream = std::cerr) const
```

Comment:
destructors are always `noexcept` in C++11 and later unless they're explicitly labeled otherwise (or they have some derived class that explicitly declares a `noexcept(false)` destructor), so this won't change anything, but I can do it.

Reply:
Implemented in 78f24549.